core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/queries/DefaultSchemaQueriesFactoryTest.java

```diff
@@ -0,0 +1,130 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.metadata.schema.queries;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.When;
+import com.datastax.oss.driver.api.core.Version;
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.internal.core.channel.DriverChannel;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class DefaultSchemaQueriesFactoryTest {
+
+  enum Expected {
+    CASS_21(Cassandra21SchemaQueries.class),
+    CASS_22(Cassandra22SchemaQueries.class),
+    CASS_3(Cassandra3SchemaQueries.class),
+    CASS_4(Cassandra4SchemaQueries.class);
+
+    final Class<? extends SchemaQueries> clz;
+
+    Expected(Class<? extends SchemaQueries> clz) {
+      this.clz = clz;
+    }
+
+    public Class<? extends SchemaQueries> getClz() {
+      return clz;
+    }
+  }
+
+  private static When<Expected> whenCassandra =
+      When.<Expected>builder()
+          .args("2.1.0")
+          .expect(Expected.CASS_21)
+          .args("2.2.0")
+          .expect(Expected.CASS_22)
+          .args("2.2.1")
+          .expect(Expected.CASS_22)
+          /* Not a real version, just documenting behaviour of existing impl */
+          .args("2.3.0")
+          .expect(Expected.CASS_22)
+          /* We now return you to real versions */
+          .args("3.0.0")
+          .expect(Expected.CASS_3)
+          .args("3.0.1")
+          .expect(Expected.CASS_3)
+          .args("3.1.0")
+          .expect(Expected.CASS_3)
+          .args("4.0.0")
+          .expect(Expected.CASS_4)
+          .args("4.0.1")
+          .expect(Expected.CASS_4)
+          .args("4.1.0")
+          .expect(Expected.CASS_4)
+          .build();
+
+  private static When<Expected> whenDse =
+      When.<Expected>builder()
+          /* DSE 6.0.0 */
+          .args("4.0.0.2284")
```

Comment:
This is failing when I run it. It returns `Expected.CASS_4`

Reply:
Yeah, this is expected.  That's the root of the underlying ticket; we're returning schema queries for DSE < 6.7.0 that don't actually match up to what that version of DSE has available.  I implemented this test as a mechanism to document and validate the underlying problem... as of this commit I haven't actually implemented the fix yet. :)